diem-move/diem-framework/experimental/sources/MultiTokenBalance.move

```diff
@@ -79,6 +130,26 @@ module DiemFramework::MultiTokenBalance {
         }
     }
 
+    public fun get_token_balance_<TokenType: store>(owner: address, token_id: GUID::ID
+    ): u64 acquires TokenBalance {
+        let gallery = &borrow_global<TokenBalance<TokenType>>(owner).gallery;
+        let index_opt = index_of_token<TokenType>(gallery, &token_id);
+
+        if (Option::is_none(&index_opt)) {
+            0
+        } else {
+            let index = Option::extract(&mut index_opt);
+            MultiToken::balance(Vector::borrow(gallery, index))
+        }
+    }
+
+    spec get_token_balance_ {
+        let gallery = get_token_balance_gallery<TokenType>(owner);
+        let min_token_id = choose min i in 0..len(gallery) where gallery[i].id == token_id;
+        ensures (forall i in 0..len(gallery): gallery[i].id != token_id) ==> result == 0;
```

Comment:
Maybe there should be a spec fun "is_in_gallery" for (exists i in 0..len(gallery): gallery[i].id != token_id)
!is_in_gallery would convert into this forall.
This would reduce verbosity and improve readability.